src/Persistence/Initialization/Updates/FixChaosMixesPlugInSeason6.cs

```diff
@@ -0,0 +1,422 @@
+
+    /// <inheritdoc />
+    public override string Name => PlugInName;
+
+    /// <inheritdoc />
+    public override string Description => PlugInDescription;
+
+    /// <inheritdoc />
+    public override string DataInitializationKey => VersionSeasonSix.DataInitialization.Id;
+
+    /// <inheritdoc />
+    public override UpdateVersion Version => UpdateVersion.FixChaosMixesSeason6;
+
+    /// <inheritdoc />
+    protected override async ValueTask ApplyAsync(IContext context, GameConfiguration gameConfiguration)
+    {
+        // Dark horse spirit and raven spirit drop item groups id fix (do this first because it persists changes to DB)
+        if (gameConfiguration.Items.Single(id => id.Name == "Spirit") is { } spirit)
+        {
+            spirit.MaximumItemLevel = 1;
+            var maps = gameConfiguration.Maps;
+            if (gameConfiguration.DropItemGroups.Single(dig => dig.Description == "Dark Horse Spirit") is { } horseGroup
+                && gameConfiguration.DropItemGroups.Single(dig => dig.Description == "Dark Raven Spirit") is { } ravenGroup)
+            {
+                ClearDropItemGroup(horseGroup);
+                ClearDropItemGroup(ravenGroup);
+                await context.SaveChangesAsync().ConfigureAwait(false);
+                CreateDropItemGroup(true);
+                CreateDropItemGroup(false);
+            }
+
+            void ClearDropItemGroup(DropItemGroup group)
+            {
+                group.PossibleItems.Clear();
+                foreach (var map in maps)
+                {
+                    if (map.DropItemGroups.FirstOrDefault(dig => dig.GetId() == group.GetId()) is { } mapDropItemGroup)
+                    {
+                        map.DropItemGroups.Remove(mapDropItemGroup);
+                    }
+                }
+
+                gameConfiguration.DropItemGroups.Remove(group);
+                context.DeleteAsync(group);
+            }
+
+            void CreateDropItemGroup(bool isHorse)
+            {
+                var group = context.CreateNew<DropItemGroup>();
+                group.SetGuid(NumberConversionExtensions.MakeWord(13, 31).ToSigned(), (short)(isHorse ? 0 : 1));
+                group.ItemLevel = (byte)(isHorse ? 0 : 1);
+                group.Chance = 0.001;
+                group.Description = $"Dark {(isHorse ? "Horse" : "Raven")} Spirit";
+                group.PossibleItems.Add(spirit);
+                group.MinimumMonsterLevel = (byte)(isHorse ? 102 : 96);
+
+                gameConfiguration.DropItemGroups.Add(group);
+                foreach (var map in maps)
+                {
+                    map.DropItemGroups.Add(group);
+                }
+            }
+        }
+
+        await base.ApplyAsync(context, gameConfiguration).ConfigureAwait(false);
```

Comment:
Can you call this at the beginning or end of the method? In the middle it looks strange 😅

Reply:
The above comment is also the reason why I left this to 2nd place in this Update Plugin, so nothing else gets done if you don't get past that first. Let me know if you still want me to move it elsewhere.